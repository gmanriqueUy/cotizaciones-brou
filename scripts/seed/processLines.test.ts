import { processLines } from "./processLines";

const line = [
  2,
  "Enero",
  2023,
  38.7,
  41.2,
  undefined,
  39.2,
  40.7,
  undefined,
  40.41,
  45.36,
  undefined,
  0.05,
  0.35,
  undefined,
  7.17,
  9.17,
];
const lastDate = new Date(0);

describe("processLines", () => {
  it("should process a single line", () => {
    expect(processLines([line], lastDate)).toMatchSnapshot();
  });
});
