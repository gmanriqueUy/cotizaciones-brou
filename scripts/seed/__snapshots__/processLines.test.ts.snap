// Vitest Snapshot v1

exports[`processLines > should process a single line 1`] = `
[
  {
    "buy": 38.7,
    "date": 2023-01-02T00:00:00.000Z,
    "iso": "USD",
    "sell": 41.2,
  },
  {
    "buy": 0.05,
    "date": 2023-01-02T00:00:00.000Z,
    "iso": "ARS",
    "sell": 0.35,
  },
  {
    "buy": 7.17,
    "date": 2023-01-02T00:00:00.000Z,
    "iso": "BRL",
    "sell": 9.17,
  },
  {
    "buy": 40.41,
    "date": 2023-01-02T00:00:00.000Z,
    "iso": "EUR",
    "sell": 45.36,
  },
]
`;
